Basic Mongo 3 support for metrics-mongodb.rb

jhsubscribe opened the pull request:
Mongo 3 serverStatus doesn't provide IndexCounters.
Fixes #15

analytically commented:
Would you also be able to upgrade the GEM to 1.12.3 and test with Mongo 2 and 3?

jhsubscribe commented:
Right.. I'm a bit of a ruby novice but I think I've done this correctly.  I've updated the gem's, built it locally (had to adjust gem signing), tested it on 2 and 3 and then committed the gem upgrades while discarding my changes to signing.

NB: I've also updated the code as I found an error on Mongo versions like 2.6.11-pre.

I think this should be ready to merge now.
